From 221d210e75a9bceb36eed039b0dea191128a66d8 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Tue, 7 Nov 2017 12:13:36 -0800
Subject: [PATCH] ni: zynq: Mender.io support for NI Ettus Research Neon Rev2
 SDR

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 configs/ni_neon_rev2_defconfig |  1 +
 include/configs/ni_neon_rev2.h | 17 ++++++++---------
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/configs/ni_neon_rev2_defconfig b/configs/ni_neon_rev2_defconfig
index a0ff5ed37a..a172c1bc8e 100644
--- a/configs/ni_neon_rev2_defconfig
+++ b/configs/ni_neon_rev2_defconfig
@@ -8,6 +8,7 @@ CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-neon-rev2"
 CONFIG_DEBUG_UART=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV"
 CONFIG_BOOTDELAY=0
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/include/configs/ni_neon_rev2.h b/include/configs/ni_neon_rev2.h
index 643b9015c6..2033f7c88a 100644
--- a/include/configs/ni_neon_rev2.h
+++ b/include/configs/ni_neon_rev2.h
@@ -9,32 +9,31 @@
 #ifndef __CONFIG_NI_NEON_REV2_H
 #define __CONFIG_NI_NEON_REV2_H
 
+#define CONFIG_ENV_IS_NOWHERE
 #include <configs/zynq-common.h>
-
-#define CONFIG_SYS_MMC_ENV_DEV 0
-#define CONFIG_ENV_SECT_SIZE           CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET              0xE0000
+#undef CONFIG_ENV_IS_NOWHERE
 
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-	"fit_image=fit.itb\0"		\
+	"fit_image=boot/fitImage\0"	\
 	"ec_compat=2\0"		\
 	"mboard=neon-rev2\0"	\
 	"ec_disable_swsync=0\0"		\
 	"ec_image=lib/firmware/ni/ec-neon-rev2.RW.bin\0"	\
 	"expand_ec_image=setenv ec_image lib/firmware/ni/ec-neon-rev${ec_compat}.RW.bin\0" \
-	"bootargs=root=/dev/mmcblk0p2 rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
+	"bootargs=root=${mender_kernel_root} rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"           \
 	"fdt_high=0x20000000\0"         \
 	"initrd_high=0x20000000\0"      \
-	"sdboot=if test ${ec_disable_swsync} = 0; then "\
+	"sdboot=run mender_setup; " \
+		"if test ${ec_disable_swsync} = 0; then "\
 			"echo Copying MCU FW from SD to RAM... && " \
 			"run expand_ec_image; " \
-			"load mmc 0 ${load_addr} ${ec_image} && " \
+			"ext4load ${mender_uboot_root} ${load_addr} ${ec_image} && " \
 			"crosec swsync ${load_addr} ${filesize}; fi;" \
 		"echo Copying FIT from SD to RAM... && " \
-		"load mmc 0 ${load_addr} ${fit_image} && " \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
 		"tftpboot ${load_addr} ${fit_image} && " \
-- 
2.14.3

